sdk/python/feast/registry.py

```diff
@@ -143,44 +154,84 @@ def clone(self) -> "Registry":
         new_registry._registry_store = NoopRegistryStore()
         return new_registry
 
-    # TODO(achals): This method needs to be filled out and used in the feast plan/apply methods.
-    @staticmethod
-    def diff_between(
-        current_registry: RegistryProto, new_registry: RegistryProto
-    ) -> RegistryDiff:
-        diff = RegistryDiff()
+    def extract_objects_for_keep_delete_update_add(
+        self, current_project: str, desired_repo_contents: RepoContents,
+    ):
+        """
+        Returns the objects that must be modified to achieve the desired repo state.
 
-        attribute_to_object_type_str = {
-            "entities": "entity",
-            "feature_views": "feature view",
-            "feature_tables": "feature table",
-            "on_demand_feature_views": "on demand feature view",
-            "request_feature_views": "request feature view",
-            "feature_services": "feature service",
+        Args:
+            current_project: The Feast project whose objects should be compared.
+            desired_repo_contents: The desired repo state.
+        """
+        objs_to_keep = {}
+        objs_to_delete = {}
+        objs_to_update = {}
+        objs_to_add = {}
+
+        registry_object_type_to_objects: Dict[str, List[Any]]
+        registry_object_type_to_objects = {
+            "entities": self.list_entities(project=current_project),
+            "feature_views": self.list_feature_views(project=current_project),
+            "on_demand_feature_views": self.list_on_demand_feature_views(
+                project=current_project
+            ),
+            "request_feature_views": self.list_request_feature_views(
+                project=current_project
+            ),
+            "feature_services": self.list_feature_services(project=current_project),
         }
 
-        for object_type in [
-            "entities",
-            "feature_views",
-            "feature_tables",
-            "on_demand_feature_views",
-            "request_feature_views",
-            "feature_services",
-        ]:
+        for object_type in REGISTRY_OBJECT_TYPES:
             (
-                objects_to_keep,
-                objects_to_delete,
-                objects_to_add,
-            ) = tag_proto_objects_for_keep_delete_add(
-                getattr(current_registry, object_type),
-                getattr(new_registry, object_type),
+                to_keep,
+                to_delete,
+                to_update,
+                to_add,
+            ) = tag_objects_for_keep_delete_update_add(
+                registry_object_type_to_objects[object_type],
+                getattr(desired_repo_contents, object_type),
             )
 
+            objs_to_keep[object_type] = to_keep
```

Comment:
What's the purpose of having these mapped by object type? Seems like the caller of this immediately flattens it back into one big list. That also seems to introduce this need to look at REGISTRY_OBJECT_TYPES

Reply:
we map these by object type because `diff_between` uses `extract_objects_for_keep_delete_update_add`, and `diff_between` needs to know the keep/delete/update/add split by object type in order to compare objects of the same type